richeditor-compose/src/commonMain/kotlin/com/mohamedrejeb/richeditor/model/RichTextState.kt

```diff
@@ -551,14 +551,28 @@ class RichTextState internal constructor(
     }
 
     fun toggleUnorderedList() {
-        val paragraph = getRichParagraphByTextIndex(selection.min - 1) ?: return
-        if (paragraph.type is UnorderedList) removeUnorderedList()
-        else addUnorderedList()
+        val firstParagraph = getRichParagraphByTextIndex(selection.min - 1) ?: return
+        val paragraphs = getRichParagraphListByTextRange(selection).toMutableList()
+        if (paragraphs.isEmpty()) {
+            paragraphs.add(firstParagraph)
+        }
+        val removeUnorderedList = firstParagraph.type is UnorderedList
+        paragraphs.forEach { paragraph ->
+            if (removeUnorderedList) {
+                removeUnorderedList(paragraph)
+            } else {
+                addUnorderedList(paragraph)
+            }
+        }
     }
 
+    @Deprecated("Use toggleUnorderedList() instead to add or remove unordered list")
     fun addUnorderedList() {
         val paragraph = getRichParagraphByTextIndex(selection.min - 1) ?: return
+        addUnorderedList(paragraph)
```

Comment:
I think this shouldn't be deprecated, but it can work the same as toggle.

- Get the list of the selected paragraphs.
- call addUnorderedList on each paragraph.

Reply:
ready